front/src/reducers/osrdconf/osrdConfCommon/index.ts

```diff
@@ -429,9 +429,27 @@ export function buildCommonConfReducers<S extends OsrdConfState>(): CommonConfRe
 
       const isInVias = isVia(pathSteps, action.payload);
       if (isInVias) {
-        state.pathSteps = replaceElementAtIndex(state.pathSteps, index, newVia);
+        // Because of import issues, there can be multiple ops with same position on path
+        // To avoid updating the wrong one, we need to find the one that matches the payload
+        const stepIndex = pathSteps.findIndex(
+          (step) =>
+            ('uic' in step &&
+              'ch' in step &&
+              step.uic === action.payload.uic &&
+              step.ch === action.payload.ch &&
+              step.name === action.payload.name) ||
+            step.id === action.payload.opId
+        );
+        newVia = { ...newVia, id: pathSteps[stepIndex].id }; // We don't need to change the id of the updated via
+        state.pathSteps = replaceElementAtIndex(state.pathSteps, stepIndex, newVia);
       } else {
-        state.pathSteps = addElementAtIndex(state.pathSteps, index, newVia);
+        // Because of import issues, there can be multiple ops at position 0
+        // To avoid inserting a new via before the origin we need to check if the index is 0
+        state.pathSteps = addElementAtIndex(
```

Comment:
You can compute index here, instead of doing it l405
```suggestion
        const index = pathSteps.findIndex(
          (step) => step.positionOnPath! >= action.payload.positionOnPath
        );
        state.pathSteps = addElementAtIndex(
```

Reply:
Good idea ! Done.